cli/src/command/generate_config.rs

```diff
@@ -104,30 +102,40 @@ impl GenerateConfig {
     }
 
     /// If no port is provided by user, uses `DEFAULT_EXONUM_LISTEN_PORT`.
-    fn parse_external_address(input: &str) -> Result<SocketAddr, Error> {
-        if let Ok(address) = input.parse() {
-            Ok(address)
-        } else {
-            let ip_address = input.parse()?;
-            Ok(SocketAddr::new(ip_address, DEFAULT_EXONUM_LISTEN_PORT))
+    fn resolve_peer_address(hostname: &str) -> Result<SocketAddr, Error> {
+        match hostname.to_socket_addrs() {
+            Ok(mut addrs) => addrs.next().ok_or_else(|| {
+                Error::msg(format!(
+                    "No one IP address is related to the domain: {}",
+                    hostname
+                ))
+            }),
+            Err(e) => {
+                if e.kind() == ErrorKind::InvalidInput {
+                    Self::resolve_peer_address(&format!(
+                        "{}:{}",
+                        hostname, DEFAULT_EXONUM_LISTEN_PORT
+                    ))
+                } else {
+                    Err(Error::from(e))
+                }
+            }
         }
     }
 
-    /// Returns `provided` address or and address combined from all-zeros
-    /// IP address and the port number from `external_address`.
-    fn get_listen_address(
-        provided: Option<SocketAddr>,
-        external_address: SocketAddr,
-    ) -> SocketAddr {
-        if let Some(provided) = provided {
-            provided
-        } else {
-            let ip_address = match external_address.ip() {
-                IpAddr::V4(_) => "0.0.0.0".parse().unwrap(),
-                IpAddr::V6(_) => "::".parse().unwrap(),
-            };
-            SocketAddr::new(ip_address, external_address.port())
-        }
+    /// Returns `provided` address or [`INADDR_ANY`](https://en.wikipedia.org/wiki/0.0.0.0) address
+    /// combined with the port number obtained from `peer_address`.
+    fn get_listen_address(provided: Option<SocketAddr>, peer_address: &str) -> SocketAddr {
+        provided.unwrap_or_else(|| match Self::resolve_peer_address(peer_address) {
```

Comment:
Maybe, it could make more sense to just attempt to get *the port* from `peer_address` (which seems pretty easy - take the last ocurrence of `:`, parse the substring to the right as `u16`) instead of trying to resolve it? I'm not sure resolution will work in all cases, and the resolved IP address seems unused anyway.

Reply:
Yes. I've tried to do so. But:

> take the last occurrence of :

There can be a problem with IPv6 addresses with such approach. 

> the resolved IP address seems unused anyway.

But we can be sure we provide a correct domain name.